// amadeus/lib/amadeus/namespaces/travel/analytics/air_traffic.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _traveled = _interopRequireDefault(require("./air_traffic/traveled.js"));

var _booked = _interopRequireDefault(require("./air_traffic/booked.js"));

var _searched = _interopRequireDefault(require("./air_traffic/searched.js"));

var _searched_by_destination = _interopRequireDefault(require("./air_traffic/searched_by_destination.js"));

var _busiest_period = _interopRequireDefault(require("./air_traffic/busiest_period.js"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

/**
 * A namespaced client for the
 * `/v1/travel/analytics/air-traffic` endpoints
 *
 * Access via the {@link Amadeus} object
 *
 * ```js
 * let amadeus = new Amadeus();
 * amadeus.travel.analytics.airTraffic;
 * ```
 *
 * @param {Client} client
 */
var AirTraffic = function AirTraffic(client) {
  _classCallCheck(this, AirTraffic);

  this.client = client;
  this.traveled = new _traveled["default"](client);
  this.booked = new _booked["default"](client);
  this.searched = new _searched["default"](client);
  this.searchedByDestination = new _searched_by_destination["default"](client);
  this.busiestPeriod = new _busiest_period["default"](client);
};

var _default = AirTraffic;
exports["default"] = _default;
module.exports = exports["default"];
